particle-services/src/app_services.rs

```diff
@@ -254,11 +257,17 @@ impl ParticleAppServices {
         config: ServicesConfig,
         modules: ModuleRepository,
         metrics: Option<ServicesMetrics>,
+        health_registry: Option<&mut HealthCheckRegistry>,
     ) -> Self {
         let vault = ParticleVault::new(config.particles_vault_dir.clone());
         let management_peer_id = config.management_peer_id;
         let builtins_management_peer_id = config.builtins_management_peer_id;
-        let this = Self {
+        let health = health_registry.map(|registry| {
+            let persisted_services = PersistedServiceHealthCheck::new();
+            registry.register("persisted_services", persisted_services.clone());
+            persisted_services
```

Comment:
Naming really confuses me here

```suggestion
            let health = PersistedServiceHealthCheck::new();
            registry.register("persisted_services", health());
            health
```